### DISCUSSION TITLE
How to handle stale connection errors correctly?

### DISCUSSION THREAD

**Discussion by LostInDarkMath** [nats-io/nats.py#860] [Q&A]
Hi there!

I registered an error callback for the NATS client and I often receive stale connection errors if the client was long time idle (which is expected).
My question is: Should I worry about this? How does this library handles this error? Does it a rreconnect and a retry? Or is my data that I try to publish lost?

I took a look as your documentation and your code but I couldn't find out how you implemention works. So I'm looking forward to hearing from you :)

**Comment by xiaofxf** [comment]
ping_interval  does not take action
